lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1327,6 +1327,7 @@ def visit_on_duplicate_key_update(self, on_duplicate, **kw):
             else:
 
                 def replace(obj):
+                    nonlocal alias_clause
                     if (
```

Comment:
**mike bayer** (zzzeek) wrote:

I can see this is trying to make it so that we dont emit the "AS _new" if there are no actual values in the on update, I think we are "safer" just emitting "AS _new" in all cases so that there is less variance in syntax and makes the code here simpler.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4412

Reply:
Makes sense, thank you!